### PR TITLE
COMMERCE-2262 Minium gallery moved to Clay and React

### PR DESCRIPTION
**Pull request by gcmznt**

### CODE REVIEW

File: commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/_gallery.scss [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    left: 0;
+  }
+
+  .next {
+    cursor: e-resize;
+    right: 0;
+    transform: rotate(180deg);
+  }
+
+  .loading-animation {
```
Comment (ethanbustad) [inline commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/_gallery.scss]: selectors should be sorted alphabetically -- I think that would be:

    .gallery-overlay
    .gallery-thumbnails
    .loading-animation
    .main-image
    .next, .prev
    .next
    .prev

File: commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/Gallery.es.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      loaded: new Set(),
+      selected: 0
+    };
+
+    this.goToPrev = this.goToPrev.bind(this);
+    this.goToNext = this.goToNext.bind(this);
+    this.loadImage = this.loadImage.bind(this);
+    this.selectImage = this.selectImage.bind(this);
+    this.openFullscreen = this.openFullscreen.bind(this);
+    this.closeFullscreen = this.closeFullscreen.bind(this);
```
Comment (ethanbustad) [inline commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/Gallery.es.js]: these should probably be alphabetized

File: commerce-frontend-js/dev/public/gallery.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+<html lang="en">
+<head>
+	<meta charset="utf-8" />
+
+	<meta
+		content="width=device-width, initial-scale=1, shrink-to-fit=no"
+		name="viewport"
+	/>
+	<!-- <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/clay/lib/css/atlas.css" />
+	-->
```
Comment (ethanbustad) [inline commerce-frontend-js/dev/public/gallery.html]: can you remove this comment? AFAIU, it doesn't add any value here (talked this over with Fabio recently on a similar PR -- feel free to ping me or him if you disagree)

**Review by ethanbustad — CHANGES_REQUESTED** [review]

File: commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/entry.dev.es.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ const props = {
     title: img,
     thumbnailUrl: getImgUrl(img, 100),
     smallUrl: getImgUrl(img, 800),
-    url: getImgUrl(img, 1600)
+    bigUrl: getImgUrl(img, 1600)
```
Comment (ethanbustad) [inline commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/entry.dev.es.js]: This should probably be sorted higher now.....

**Review by ethanbustad — APPROVED** [review]
Just one more alphabetizing change to make -- after that, I'll stop complaining :)

**Comment by gcmznt** [comment]
ci:test

**Comment by gcmznt** [comment]
ci:test

**Comment by gcmznt** [comment]
ci:test

**Comment by gcmznt** [comment]
ci:test

**Comment by gcmznt** [comment]
@marco-leo I cannot understand why this build fails. What can I do to make it pass?

**Comment by ethanbustad** [comment]
Sorry, @gcmznt , this PR does not look like a fun time :(. Looking at the `For full console, click here.` link, I think this is the relevant error message:

     /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/Gallery.es.js
     142:4  error  Expected object keys to be in natural ascending order. 'thumbnailUrl' should be before 'title'     sort-keys
     143:4  error  Expected object keys to be in natural ascending order. 'smallUrl' should be before 'thumbnailUrl'  sort-keys

     /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/MainImage.js
     37:2  error  Expected object keys to be in natural ascending order. 'src' should be before 'title'  sort-keys

     /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/Thumbnail.js
     36:2  error  Expected object keys to be in natural ascending order. 'src' should be before 'title'  sort-keys

     /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/Thumbnails.js
     32:4  error  Expected object keys to be in natural ascending order. 'thumbnailUrl' should be before 'title'  sort-keys

     /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/js/gallery/entry.dev.es.js
     19:3  error  Expected object keys to be in natural ascending order. 'thumbnailUrl' should be before 'title'     sort-keys
     20:3  error  Expected object keys to be in natural ascending order. 'smallUrl' should be before 'thumbnailUrl'  sort-keys

**Comment by gcmznt** [comment]
ci:test

Thanks @ethanbustad

**Comment by marco-leo** [comment]
@FabioDiegoMastrorilli @gcmznt How does the css loading works?  who will load that css in the page?

**Pull request merged**